code/modules/modular_computers/file_system/programs/generic/configurator.dm

```diff
@@ -35,6 +66,30 @@
 		data["battery_rating"] = battery_module.battery.maxcharge
 		data["battery_percent"] = round(battery_module.battery.percent())
 
+	var/obj/item/stock_parts/computer/network_card/network_card = program.computer.get_component(PART_NETWORK)
+	data["nic_exists"] = !!network_card
+	if(network_card)
+		var/datum/extension/exonet_device/exonet = get_extension(network_card, /datum/extension/exonet_device)
+		var/datum/exonet/network = exonet.get_local_network()
+		if(!network)
+			data["signal_strength"] = "Not Connected"
+		else
+			var/signal_strength = network.get_signal_strength(network_card, network_card.get_netspeed())
+			if(signal_strength <= 0)
+				data["signal_strength"] = "Not Connected"
+			else if(signal_strength <= 6)
+				data["signal_strength"] = "Low Signal"
+			else
+				data["signal_strength"] = "High Signal"
+		if(network_card.ennid)
```

Comment:
I think these params should be pulled out of the extension instead, these seems to be this weird duplication of state between objects and extension they hold